docs/backend-system/building-plugins-and-modules/08-migrating.md

```diff
@@ -73,128 +73,102 @@ export const kubernetesPlugin = createBackendPlugin({
 });
 ```
 
-Done! Users of this plugin are now able to import the `kubernetesPlugin` and register it in their backend using
+Lastly, make sure you re-export the plugin instance as the default export of your package in `src/index.ts`:
+
+```ts
+export { kubernetesPlugin as default } from './plugin.ts';
+```
+
+Done! Users of this plugin are now able to import your plugin package and register it in their backend using
 
 ```ts
 // packages/backend/src/index.ts
-import { kubernetesPlugin } from '@backstage/plugin-kubernetes-backend';
-backend.add(kubernetesPlugin);
+backend.add(import('@backstage/plugin-kubernetes-backend'));
 ```
 
 There's one thing missing that those sharp eyed readers might have noticed: the `clusterSupplier` option is missing from the original plugin. Let's add it and discuss the alternatives.
 
-One alternative is to pass the `ClusterSupplier` in as options to the plugin, which is quick and easy but not very flexible, and also hard to evolve without introducing breaking changes as it changes the public API for the plugin. Having complex types passed in directly to the plugin also clutters the backend setup code and makes it harder to read.
-
-Options are primarily used for simple configuration values that are not complex types. In this case we want to allow users to register their own `ClusterSupplier` implementations to the plugin. This is where the new backend system's [extension points](../architecture/05-extension-points.md) come in handy, but let's look at doing this with options first.
+One alternative is to make it possible to build the cluster supplier using static configuration. It could for example be that there are a selection of built-in implementations to choose from, or that the logic for how the `ClusterSupplier` is supposed to function is all determined by configuration, or a combination of the two. Using static configuration for customization is always the preferred option whenever it's possible. In this case, we could for example imagine that we would be able to configure our cluster supplier like this:
```

Comment:
```suggestion
One alternative is to make it possible to build the cluster supplier using static configuration. It could for example be that there is a selection of built-in implementations to choose from, or that the logic for how the `ClusterSupplier` is supposed to function is all determined by configuration, or a combination of the two. Using static configuration for customization is always the preferred option whenever it's possible. In this case, we could for example imagine that we would be able to configure our cluster supplier like this:
```